x-pack/plugins/apm/server/routes/service_groups/get_services_counts.ts

```diff
@@ -7,49 +7,67 @@
 
 import { ProcessorEvent } from '@kbn/observability-plugin/common';
 import { rangeQuery, kqlQuery } from '@kbn/observability-plugin/server';
+import { QueryDslQueryContainer } from '@elastic/elasticsearch/lib/api/types';
 import { Setup } from '../../lib/helpers/setup_request';
 import { SERVICE_NAME } from '../../../common/elasticsearch_fieldnames';
+import { SavedServiceGroup } from '../../../common/service_groups';
 
 export async function getServicesCounts({
   setup,
-  kuery,
-  maxNumberOfServices,
   start,
   end,
+  serviceGroups,
 }: {
   setup: Setup;
-  kuery: string;
-  maxNumberOfServices: number;
   start: number;
   end: number;
+  serviceGroups: SavedServiceGroup[];
 }) {
   const { apmEventClient } = setup;
 
-  const response = await apmEventClient.search('get_services_count', {
+  const serviceGroupsKueryMap: Record<string, QueryDslQueryContainer> =
+    serviceGroups.reduce((acc, sg) => {
+      return {
+        ...acc,
+        [sg.id]: kqlQuery(sg.kuery)[0],
```

Comment:
I think you need to wrap this in a bool clause rather than pick the first item from the list (not sure if it's guaranteed that there's only one item in the list).

Reply:
the `kqlQuery` does  produce a query object which includes `bool` at the top level:

<details><summary>Example</summary>
<p>

```json
    "aggs": {
      "service_groups": {
        "filters": {
          "filters": {
            "cad53530-3ab7-11ed-a460-51ceee1b0a2c": {
              "bool": {
                "should": [
                  {
                    "query_string": {
                      "fields": ["service.name"],
                      "query": "opbeans*"
                    }
                  }
                ],
                "minimum_should_match": 1
              }
            },
            "086c1d50-3ab8-11ed-a460-51ceee1b0a2c": {
              "bool": {
                "should": [{ "match": { "agent.name": "python" } }],
                "minimum_should_match": 1
              }
            },
            "fc75d400-3ab7-11ed-a460-51ceee1b0a2c": {
              "bool": {
                "should": [{ "match": { "agent.name": "java" } }],
                "minimum_should_match": 1
              }
            },
            "16e4cbc0-3ab8-11ed-a460-51ceee1b0a2c": {
              "bool": {
                "should": [{ "match": { "agent.name": "nodejs" } }],
                "minimum_should_match": 1
              }
            },
            "308aa720-3ab8-11ed-a460-51ceee1b0a2c": {
              "bool": {
                "should": [{ "match": { "agent.name": "go" } }],
                "minimum_should_match": 1
              }
            },
            "59164aa0-3ab8-11ed-a460-51ceee1b0a2c": {
              "bool": {
                "should": [{ "match_phrase": { "agent.name": "rum-js" } }],
                "minimum_should_match": 1
              }
            },
            "3efb8d60-3ab8-11ed-a460-51ceee1b0a2c": {
              "bool": {
                "should": [
                  {
                    "query_string": {
                      "fields": ["service.name"],
                      "query": "*beat"
                    }
                  }
                ],
                "minimum_should_match": 1
              }
            }
          }
        },
        "aggs": {
          "services_count": { "cardinality": { "field": "service.name" } }
        }
      }
    }
```

</p>
</details>